android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/FormUtils.java

```diff
@@ -1725,6 +1725,15 @@ protected void extractOptionOpenMRSAttributes(JSONArray valueOpenMRSAttributes,
             valueOpenMRSObject.put(JsonFormConstants.OPENMRS_ENTITY_PARENT, openmrsEntityParent);
             valueOpenMRSObject.put(JsonFormConstants.OPENMRS_ENTITY, openmrsEntity);
             valueOpenMRSObject.put(JsonFormConstants.OPENMRS_ENTITY_ID, openmrsEntityId);
+            if(itemOption.has(JsonFormConstants.TEXT) && itemOption.has(JsonFormConstants.KEY)) {
+                valueOpenMRSObject.put(JsonFormConstants.VALUE, itemOption.getString(JsonFormConstants.KEY));
+                valueOpenMRSObject.put(JsonFormConstants.TEXT, itemOption.getString(JsonFormConstants.TEXT));
+                JSONArray options = new JSONArray();
+                options.put(itemOption);
+                // adds the selected option as an option array to fill the human readable value;
+                valueOpenMRSObject.put(JsonFormConstants.OPTIONS_FIELD_NAME,options);
+            }
```

Comment:
Do we have a situation where the key and Text are null? Or are we using the `.has` to avoid Null pointers?

Reply:
yes @dubdabasoduba  we are Using .has to avoid null pointers because some widgets like date picker will have value and no text